lib/service.go

```diff
@@ -225,15 +236,25 @@ func (m *RelayService) GetPayloadHeaderV1(r *http.Request, args *string, result
 		log.Println("GetPayloadHeaderV1: no TransactionsRoot found, calculating it from Transactions list instead: ", *args, result.BlockHash, result.Number)
 
 		txs := types.Transactions{}
+		sszTxs := TransactionsSSZ{}
 		for i, otx := range *result.Transactions {
 			var tx types.Transaction
-			if err := tx.UnmarshalBinary(common.Hex2Bytes(otx)); err != nil {
+			bytesTx := common.Hex2Bytes(otx)
+			if err := tx.UnmarshalBinary(bytesTx); err != nil {
 				log.Println("GetPayloadHeaderV1: error decoding tx: ", err)
 				return fmt.Errorf("failed to decode tx %d: %v", i, err)
 			}
 			txs = append(txs, &tx)
+			sszTxs.Transactions = append(sszTxs.Transactions, bytesTx)
 		}
-		newRoot := types.DeriveSha(txs, trie.NewStackTrie(nil))
+
+		newRootBytes, err := sszTxs.HashTreeRoot()
```

Comment:
I recall there's a bug with this. I will check with the fssz implementers. I had handcraft my own to comply on kintsugi:

See: https://github.com/prysmaticlabs/prysm/pull/9853/files#diff-7f3786d768dbf8947d4e60b75eb8ed9f2e678f41e5a60c491c0b994fa4ab745dR103

Reply:
This is so close to perfect for me, but I'm having issues when importing `github.com/prysmaticlabs/prysm`:
```
../../.go/pkg/mod/github.com/prysmaticlabs/prysm@v1.4.2-0.20211211173600-d6338f60427f/proto/prysm/v1alpha1/generated.ssz.go:4033:7: hh.AppendBytes32 undefined (type *ssz.Hasher has no field or method AppendBytes32)
```

happens for both master and kintusgi branches of prysm. I'm going to go ahead and vendor the code for now, so thanks for pointing it out!